app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -171,10 +171,15 @@ def setup_agent(instance)
     # prepare work directory
     ssh.perform_commands(["sudo mkdir -p #{WORK_DIR}"])
     ssh.perform_commands(["sudo chmod go+w #{WORK_DIR}"])
+
     # scp the default setting yaml file
-    Tempfile.open('config.yml') do |config|
```

Comment:
why was this change made?

Reply:
Two reasons: 1. need delete the temp file; 2. the generate temp file just prefix with name I given, need right name in the remote side.

Reply:
1. isn't the temp file deleted when you leave the `Tempfile.open('config.yml') do |config|` block?
2. The name of the temp file on the appliance doesn't need to have any relation to the name on the remote side. You just need to scp it to the name you want.

Reply:
1. No, I thought it too, but the block only sure to close it. 
2. Yes, now I give the exact name in remote side.